src/etc/inc/unbound.inc

```diff
@@ -518,13 +519,25 @@ function unbound_add_host_entries() {
             if ($host['host'] != "")
                 $host['host'] = $host['host'].".";
             if (!$added_item[$current_host]) {
-                $host_entries .= "local-data-ptr: \"{$host['ip']} {$host['host']}{$host['domain']}\"\n";
-                if (is_ipaddrv6($host['ip']))
-                    $host_entries .= "local-data: \"{$host['host']}{$host['domain']} IN AAAA {$host['ip']}\"\n";
-                else
-                    $host_entries .= "local-data: \"{$host['host']}{$host['domain']} IN A {$host['ip']}\"\n";
-                if (!empty($host['descr']) && isset($config['unbound']['txtsupport']))
+                /* Backwards compatibility for records created before introducing RR types. */
+                if (!isset($host['rr'])) {
+                    $host['rr'] = (is_ipaddrv6($host['ip'])) ? 'AAAA' : 'A';
```

Comment:
Here you're using "AAAA", but later you're using only "A" for both. If we do resource records, we ought to split A and AAAA.